src/main/java/com/google/devtools/build/lib/bazel/repository/starlark/StarlarkRepositoryContext.java

```diff
@@ -392,6 +392,61 @@ public boolean delete(Object pathObject, StarlarkThread thread)
     }
   }
 
+  @StarlarkMethod(
+      name = "rename",
+      doc = "Renames the file or directory from `src` to `dst`.",
+      useStarlarkThread = true,
+      parameters = {
+        @Param(
+            name = "src",
+            allowedTypes = {
+              @ParamType(type = String.class),
+              @ParamType(type = Label.class),
+              @ParamType(type = StarlarkPath.class)
+            },
+            doc = "The path of the existing file or directory to rename."),
+        @Param(
+            name = "dst",
+            allowedTypes = {
+              @ParamType(type = String.class),
+              @ParamType(type = Label.class),
+              @ParamType(type = StarlarkPath.class)
+            },
+            doc = "The new name of the file or directory."),
```

Comment:
Could you add that parent directories are created as needed? Please also document what happens if the target already exists.

Reply:
Done.